feat: Review crawl without analysis run

SuaYoo opened the pull request:
Resolves https://github.com/webrecorder/browsertrix/issues/2718

## Changes

Enables manual QA review for successfully finished crawls.

## Manual testing

1. Log in as crawler
2. Click "Crawling"
3. Click complete or stopped workflow
4. Click "+ Add Review" in "Latest Crawl" tab. Verify navigation to review page
5. Click pages in page list. Verify screenshot updates as expected
6. Approve, reject, and leave comments on pages. Verify pages update as expected
7. Sort by "Approval". Verify pages update as expected
8. Click "Finish Review"
9. Submit review. Verify navigation back to workflow with updated QA Rating
10. Regression test QA for a crawl with an analysis run

## Screenshots

| Page | Image/video |
| ---- | ----------- |
| QA Review | <img width="1470" height="675" alt="Screenshot 2025-07-14 at 12 05 11 PM" src="https://github.com/user-attachments/assets/fe444f36-8eb3-43eb-bdb9-6558b910d8e9" /> |
| QA Review | <img width="386" height="125" alt="Screenshot 2025-07-14 at 12 05 15 PM" src="https://github.com/user-attachments/assets/05ff9fcd-76eb-470c-968a-3b39f6e27f99" /> |
| QA Review | <img width="856" height="101" alt="Screenshot 2025-07-14 at 12 10 36 PM" src="https://github.com/user-attachments/assets/31ac771d-abc0-4e8a-b2d4-ff0b7a70363e" /> |


## Follow-ups

We should update the user guide and any related resources https://github.com/webrecorder/browsertrix/issues/2730

SuaYoo committed:
show pages

SuaYoo committed:
enable page review

SuaYoo committed:
update links

SuaYoo committed:
remove sort filter

SuaYoo committed:
hide comparison views

SuaYoo committed:
update action menu

SuaYoo commented on frontend/src/pages/org/archived-item-qa/archived-item-qa.ts:
```diff
@@ -535,6 +539,23 @@ export class ArchivedItemQA extends BtrixElement {
+              () => html`
+                <btrix-popover
+                  content=${msg(
+                    "Screenshot, text, and resource comparison views are only available for analyzed crawls. Run QA analysis from the crawl detail page to enable comparisons.",
```
@DaleLore Suggestions on how to best word this?

SuaYoo committed:
fix approval not updating page

SuaYoo committed:
update navigation

tw4l commented:
In general, this is looking nice and working well, but I think will require a little more work to make the scenario where users actually do want to run the QA analysis a bit easier/better.

Take the scenario of a new user who clicks on the "Add Review" link from the workflow after a crawl completes. Not being already familiar with our QA system, they'd only know that they can add additional information into the Review Dashboard by running analysis if they dig into a tooltip's text. If they do see that text, there's no button in the Review Dashboard to then go ahead and do that, and no easy link to direct them to the page where they'd have to kick that off. They'd only see a "Return to Workflow" link and "Exit Review", which also takes them back to the workflow. And then they'd have to know to navigate to the crawl object's QA tab on their own.

it might be that we could address this by adding a button to run Analysis directly in the Review Dashboard? That might require some additional UI changes though, e.g. to see the status of the run. Alternatively, we could provide a link to the crawl detail QA tab so that they can do it there, but making users go out to a different page, hit a button, and then come back also feels a little clunky.

In addition, I think we may need to update the docs a bit to reflect the changes made in this PR.

SuaYoo commented:
> it might be that we could address this by adding a button to run Analysis directly in the Review Dashboard? That might require some additional UI changes though, e.g. to see the status of the run. Alternatively, we could provide a link to the crawl detail QA tab so that they can do it there, but making users go out to a different page, hit a button, and then come back also feels a little clunky.

We show this run dropdown for crawls with analysis runs:

<img width="714" height="93" alt="Screenshot 2025-07-10 at 10 06 37 AM" src="https://github.com/user-attachments/assets/454eb290-a5ea-43e1-820c-62fd4d042843" />

We could add a "Run/Rerun Analysis" option here?

tw4l commented:
> We could add a "Run/Rerun Analysis" option here?

That seems like a good start! And maybe a link to the crawl's QA pane somewhere?

SuaYoo committed:
fix colors

SuaYoo committed:
run analysis from review

emma-sg commented:
I feel like "+ Add Review" ends up being a little confusing, I'd expect it to immediately let you add a review/rating, rather than taking you to a separate review page for you to then do the reviewing. Maybe "+ Review Crawl" would make more sense?

emma-sg commented:
Also I think the note about "comparison view unavailable" is a useful one to leave in maybe?

SuaYoo committed:
update labels

SuaYoo committed:
show info while running

SuaYoo commented:
> Maybe "+ Review Crawl" would make more sense?
> Also I think the note about "comparison view unavailable" is a useful one to leave in maybe?

Updated!

SuaYoo commented on frontend/src/pages/org/archived-item-qa/archived-item-qa.ts:
```diff
@@ -535,6 +552,62 @@ export class ArchivedItemQA extends BtrixElement {
+                      class="flex items-center gap-1.5 text-xs text-neutral-500"
+                    >
+                      <sl-icon class="text-sm" name="info-circle"></sl-icon>
+                      ${msg("Limited view")}
```
Open to suggestions on how to label and word this (cc @DaleLore )

<img width="386" height="125" alt="Screenshot 2025-07-14 at 12 05 15 PM" src="https://github.com/user-attachments/assets/8fea1653-f762-45b2-9ee4-ca2cc17c2011" />

emma-sg replied:
Might include "comparison" in a couple places? i.e. 

> Screenshot comparison, text comparison, and resource quality metrics are only available for analyzed crawls. Run analysis to view and compare all QA metrics.

SuaYoo committed:
update copy for finish step

SuaYoo commented:
> That seems like a good start! And maybe a link to the crawl's QA pane somewhere?

Updated with segmented "Finish" button:

<img width="230" height="154" alt="Screenshot 2025-07-15 at 2 56 32 PM" src="https://github.com/user-attachments/assets/7ba1a950-f332-4d46-84a1-51891c0187ff" />

ikreymer commented:
Looking pretty good!
Does it make sense to also add a text and resources for consistency, but without the comparison?
The resources might be useful, eg. to see if there's a video, etc.. loaded?

SuaYoo committed:
show text

SuaYoo committed:
show resources

SuaYoo committed:
update copy

SuaYoo committed:
fix non-diff resource table

SuaYoo commented:
> Does it make sense to also add a text and resources for consistency, but without the comparison? The resources might be useful, eg. to see if there's a video, etc.. loaded?

Updated:
<img width="849" height="557" alt="Screenshot 2025-07-16 at 3 40 27 PM" src="https://github.com/user-attachments/assets/c5f3d989-778b-4698-bfb5-2a3540f11f93" />

<img width="847" height="554" alt="Screenshot 2025-07-16 at 3 40 23 PM" src="https://github.com/user-attachments/assets/579f9efc-f8cc-434d-835c-5253f56cd27b" />

ikreymer reviewed (approved):
Looks good! Definitely makes it easier to review crawls w/o having to run automated QA!

tw4l commented:
A small observations from testing.

It seems like at certain screen sizes, some of the review pane options become hidden. Might want to change the layout at a larger breakpoint than it currently does?

<img width="1192" height="806" alt="Screenshot 2025-07-21 at 11 09 09 AM" src="https://github.com/user-attachments/assets/ab42dc9d-c62d-4f41-9f3c-3403b39ded8b" />

<img width="1021" height="806" alt="Screenshot 2025-07-21 at 11 09 29 AM" src="https://github.com/user-attachments/assets/79870b99-3006-476c-84b5-50ff73756789" />


Not the biggest deal, but another small regression: it seems like when QA analysis is started from the review pane, the crawl's QA tab does not display the progress/results bar for text/screenshot matches until the analysis run completes, which is different than if it is started from the QA tab:

<img width="1299" height="799" alt="Screenshot 2025-07-21 at 11 10 35 AM" src="https://github.com/user-attachments/assets/128cbc92-8539-47f0-991c-a668fbfcd93e" />

vs

<img width="1297" height="768" alt="Screenshot 2025-07-21 at 11 15 04 AM" src="https://github.com/user-attachments/assets/79c35abc-ae3b-4d38-b903-2151f8ccd07c" />

SuaYoo committed:
fix default qa run id not setting

SuaYoo committed:
adjust for smaller screens

SuaYoo commented:
> It seems like at certain screen sizes, some of the review pane options become hidden. Might want to change the layout at a larger breakpoint than it currently does?

Adjusted so that the run button is always visible:

<img width="985" height="660" alt="Screenshot 2025-07-22 at 11 40 25 AM" src="https://github.com/user-attachments/assets/3eda2f75-61c2-4ddd-b20e-5b1088454c68" />


> Not the biggest deal, but another small regression: it seems like when QA analysis is started from the review pane, the crawl's QA tab does not display the progress/results bar for text/screenshot matches until the analysis run completes, which is different than if it is started from the QA tab:

Fixed!

emma-sg reviewed (approved):
Sweet, this is looking good! Everything seems to be working.
